lib/runtime2/copilot/index.ts

```diff
@@ -0,0 +1,183 @@
+  initialDom: boolean = false;
+
+  browsingContext: Transmute.BrowsingContext;
+  memoryDocument: Document | null = null; // Store the memory DOM document
+  performanceTracker: PerformanceTracer; // Add performance tracker property
+
+  constructor(browsingContext: Transmute.BrowsingContext) {
+    this.browsingContext = browsingContext;
+    this.performanceTracker = new PerformanceTracer(); // Initialize tracker
+  }
+
+  public async creatTask(input: string) {
+    // create html dom
+    if (!this.initialDom) {
+      this.memoryDocument = this.browsingContext.start(htmlText, 'text/html');
+      this.initialDom = true;
+    }
+    this.performanceTracker.start('totalTask'); // Stop total task timer
+
+    const controller = create(new PerformanceTracer());
+    controller.on('append', (data) => {
+      // const randomColor = '#' + Math.floor(Math.random() * 16777215).toString(16);
+      // this.memoryDocument.body.style.backgroundColor = "rgba(255, 255, 255,0.2)";
+      const { type, fragment } = data;
+      const { id, content, parentId } = fragment;
+      console.log('Agent: Processed append key:', type, 'data:', fragment);
+      switch (type) {
+        case 'html':
+          this.handleAppendHtml(parentId, content);
+          break;
+        case 'css':
+          this.handleAppendCss(fragment);
+        case 'header':
+          if (!content) {
+            console.warn('Agent: Processed append key:', type, 'data:', fragment);
+            return;
+          }
+          const headerCssfragment = `#${APP_ROOT_ID}{${content}}`;
+          console.log('Agent: Processed  Header append CSS:', headerCssfragment);
+          this.handleAppendCss(headerCssfragment);
+          break;
+        case 'moudle':
+          const moudleHtmlfragment = `<div id=\"${id}\"></div>`;
+          const moudleCssfragment = `#${id}{${content}}`;
+          this.handleAppendHtml(APP_ROOT_ID, moudleHtmlfragment);
+          console.log('Agent: Processed moudle append CSS:', moudleCssfragment);
+          this.handleAppendCss(moudleCssfragment);
+          break;
+        default:
+          console.warn('Agent: Processed append key:', type, ' data:', fragment);
+          break;
+      }
+      console.log('Agent: Processed append key:', type, '  data:', fragment);
+    })
+
+    try {
+      await controller.generatePageStream(input);
+    } catch (error) {
+      console.error('Agent: Error creating task:', error);
+    }
+
+    this.performanceTracker.end('totalTask'); // Stop total task timer
+    this.performanceTracker.report();
+
+    console.log('Agent: Final HTML:', this.memoryDocument.body.innerHTML);
+    console.log('Agent: Final CSS:', this.memoryDocument.head.innerHTML);
+    const htmlcontext = `<html><head>${this.memoryDocument.head.innerHTML}</head><body>${this.memoryDocument.body.innerHTML}</body></html>`
+    this.saveHtmlToFile(htmlcontext, input);
+    return controller;
+
+  }
+
+  private handleAppendCss(content: string) {
```

Comment:
```suggestion
  #handleAppendCss(content: string) {
```

1. 使用 ES 的 Private Fields 来实现私有函数
2. 我建议把文档生成当做 Threepio 的一个能力（即需要单独抽象到一个类），append css html 等是归属于这个能力类的一个方法